frontends/concrete-python/examples/sha1/sha1.md

```diff
@@ -0,0 +1,307 @@
+
+Then, we have the main loop of the compression function:
+
+```
+    for i in range(80):
+        if 0 <= i <= 19:
+
+            # Do f = d ^ (b & (c ^ d))
+            fsplit_enc = my_module.iftern.run(bsplit_enc, csplit_enc, dsplit_enc)
+
+            ksplit = split(0x5A827999)
+        elif 20 <= i <= 39:
+
+            # Do f = b ^ c ^ d
+            fsplit_enc = my_module.xor3.run(bsplit_enc, csplit_enc, dsplit_enc)
+
+            ksplit = split(0x6ED9EBA1)
+        elif 40 <= i <= 59:
+
+            # Do f = (b & c) | (b & d) | (c & d)
+            fsplit_enc = my_module.maj.run(bsplit_enc, csplit_enc, dsplit_enc)
+
+            ksplit = split(0x8F1BBCDC)
+        elif 60 <= i <= 79:
+
+            # Do f = b ^ c ^ d
+            fsplit_enc = my_module.xor3.run(bsplit_enc, csplit_enc, dsplit_enc)
+
+            ksplit = split(0xCA62C1D6)
+```
+
+In this main loop, we take the right choice of `f` and `k`. Here, we can see a first use of
+the different functions in the Module.
+
+Then we continue with other functions in the module, to compute `arot5 = _left_rotate(a, 5)` and
+`arot5 + f + e + k + w[i]`:
+
+```
+        # Do arot5 = _left_rotate(a, 5)
+        arot5split_enc = my_module.rotate5.run(asplit_enc)
+
+        # Do arot5 + f + e + k + w[i]
+        ssplit_enc = my_module.add5.run(
+            arot5split_enc,
+            fsplit_enc,
+            esplit_enc,
+            wsplit_enc[i],
+            my_module.rotate5.encrypt(ksplit),  # BCM: later remove the encryption on k
+        )
+```
+
+Finally, in
+
+```
+        # Final update of the a, b, c, d and e registers
+        newasplit_enc = ssplit_enc
+
+        esplit_enc = dsplit_enc
+        dsplit_enc = csplit_enc
+
+        # Do c = _left_rotate(b, 30)
+        csplit_enc = my_module.rotate30.run(bsplit_enc)
+
+        bsplit_enc = asplit_enc
+        asplit_enc = newasplit_enc
+```
+
+we update the different `a, b, c, d, e` values as in the clear
+implementation but with the encrypted forms.
+
+One can remark that we have compiled the different functions of the Module on inputset made with
+bits. Under the hood, Concrete will add a few programmable bootstrapping, to compute the right
+functions in FHE.
+
+## MLIR code
+
+Compiling with `show_mlir = True` allows to see the different MLIR implementations.
+
+## Testing or Using
+
+Once again, this tutorial is more to show the use of Modules than using it in production. For a full
+client-server API, one may want to do more in FHE, including the message expansion, and maybe
+optimize more the functions.
+
+One can check that the implementation works in FHE by running `python sha1.py --autotest`: it will
+pick a certain number of random inputs, hash them in FHE and compare the result with the `hashlib`
+standard implementation.
+
+One can also hash a given value with
+`echo -n "The quick brown fox jumps over the lazy dog" | python sha1.py`, and it will print
+something like:
```

Comment:
> it will print something like:

Chang to:
it will print the following results:

Reply:
it's not exactly the same thing, since there is randomness here